hy/compiler.py

```diff
@@ -370,19 +386,25 @@ def checker(self, expression):
     return _dec
 
 
+def eq_sym(x, sym):
+    return isinstance(x, HySymbol) and x == HySymbol(sym)
+
+
+def eq_kw(x, kw):
+    return isinstance(x, HyKeyword) and x == HyKeyword(kw)
```

Comment:
You shouldn't need this, `HyKeyword.__eq__` should do the trick, unless I did something wrong.